src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,164 @@
+package seedu.duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Deals with understanding user commands.
+ */
+public class Parser {
+
+    public Parser() {};
+
+    /**
+     * Parses the user commands.
+     *
+     * @param command user command
+     * @param todolist the current ToDoList
+     * @param storage the storage associated with the current ToDoList
+     * @return isBye boolean indicated the ending 'bye' command
+     */
+    public boolean parse(String command, ToDoList todolist, Storage storage) {
```

Comment:
I have noticed that you named your `ToDoList` objects as `todolist` in this line and other lines. Should the `todolist` variable rename to `toDoList`?
```suggestion
    public boolean parse(String command, ToDoList toDoList, Storage storage) {
```

Reply:
I believe that this depends on whether `todolist` is itself a Noun. However, I can see why it may cause confusion. Good catch!